src/openms/include/OpenMS/ANALYSIS/OPENSWATH/PeakIntegrator.h

```diff
@@ -0,0 +1,386 @@
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Douglas McCloskey, Pasquale Domenico Colaianni $
+// $Authors: Douglas McCloskey, Pasquale Domenico Colaianni $
+// --------------------------------------------------------------------------
+
+#ifndef OPENMS_ANALYSIS_OPENSWATH_PEAKINTEGRATOR_H
+#define OPENMS_ANALYSIS_OPENSWATH_PEAKINTEGRATOR_H
+
+#include <OpenMS/config.h> // OPENMS_DLLAPI
+#include <OpenMS/CONCEPT/LogStream.h>
+#include <OpenMS/DATASTRUCTURES/DefaultParamHandler.h>
+#include <OpenMS/DATASTRUCTURES/String.h>
+#include <OpenMS/KERNEL/MSChromatogram.h>
+#include <OpenMS/KERNEL/MSSpectrum.h>
+
+namespace OpenMS
+{
+  /**
+    @brief Compute the area, background and shape metrics of a peak.
+
+    The containers supported by the methods are MSChromatogram and MSSpectrum.
+  */
+  class OPENMS_DLLAPI PeakIntegrator :
+    public DefaultParamHandler
+  {
+public:
+    /// Constructor
+    PeakIntegrator();
+    /// Destructor
+    virtual ~PeakIntegrator();
+
+    /**
+      @brief Estimate the background of a peak contained in a MSChromatogram.
+
+      The user can choose to compute one of two background types: "vertical_sum" and "base_to_base".
+      For the former case, the area is computed as a rectangle with delta RT being the base and
+      the minimum intensity on boundaries as the height.
+      For the latter case, the area is computed as a rectangle trapezoid. Similar to the "vertical_sum"
+      solution, this technique also takes into account the area between the intensities on boundaries.
+
+      For both cases, the parameter integration_type_ decides which formula to use to compute the area.
+      The user should make sure to use the same integration_type between calls of estimateBackground() and
+      integratePeak().
+
+      @note Make sure the chromatogram is sorted with respect to retention time.
+
+      @param[in] chromatogram The chromatogram which contains the peak
+      @param[in] left The left retention time boundary
+      @param[in] right The right retention time boundary
+    */
+    void estimateBackground(const MSChromatogram& chromatogram, const double& left, const double& right);
```

Comment:
would it make sense to also provide these functions for a start and end iterator?